frontend/src/types/oauth.ts

```diff
@@ -25,15 +58,17 @@ export function redirectUrl(): string {
   return `${window.location.origin}/oauth/callback`;
 }
 
-type OAuthType = "login" | "register";
+// login: users try to login via oauth
+// register: users try to bind a vcs to her workspace
+export type OAuthType = "login" | "register-vcs" | "link-vcs-repository";
```

Comment:
Related to the comment in getOAuthEventName.

So maybe we could just simplify it a bit to just declare 

export type OAuthType = "bb.oauth.login" | "bb.oauth.register-vcs" | "bb.oauth.link-vcs-repository";

and just pass string like "bb.oauth.login" to the eventlistener